File: C++/Medium/Accepted/number-of-substrings-containing-all-three-characters_5.cpp
class Solution {
public:
    int numberOfSubstrings(string s) {
        
        int n = s.size();
        int i = 0, j = 0, ans = 0;
        vector<int> cnt(3, 0);
        while (j < n) {
            cnt[s[j++] - 'a']++;
            while (i < j && cnt[0] && cnt[1] && cnt[2]) {
                ans += n - j + 1;
                cnt[s[i++] - 'a']--;
            }
        }
        return ans;
    }
};

File: Rust/Medium/WrongAnswer/number-of-substrings-containing-all-three-characters_5.rs
impl Solution {
    pub fn number_of_substrings(s: String) -> i32 {
        
        let mut count = [0; 3];
        let mut res = 0;
        let mut j = 0;
        let mut k = 0;
        for (i, c) in s.as_bytes().iter().enumerate() {
            count[*c as usize - 97] += 1;
            while count[0] > 0 && count[1] > 0 && count[2] > 0 {
                count[s.as_bytes()[j] as usize - 97] -= 1;
                j += 1;
            }
            k = i + 1;
            res += k - j;
        }
        res as i32
    }
}